File: backend/src/main/java/com/ums/ums_backend/entity/Student.java
package com.ums.ums_backend.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.util.List;

/**
 * Represents a student enrolled in the university with personal details and course enrollments.
 */
@Entity
@Getter
@Setter
public class Student {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne
    @JoinColumn(name="user_id", nullable = false, unique = true)
    private User user;

    @Column(nullable = false, length = 30)
    private String firstName;

    @Column(nullable = false, length = 30)
    private String lastName;

    @Enumerated(EnumType.STRING)
    private Gender gender;

    @Column(nullable = false)
    private LocalDate dateOfBirth;

    @Column(unique = true)
    private String email;

    @OneToMany(mappedBy = "student")
    private List<Enrollment> Enrollments;

}

File: frontend/src/app/features/student/models/student.ts
import {Gender} from '../../../core/enums/Gender';
import {Enrollment} from '../../enrollment/models/enrollment';

export interface Student {

  id: number,
  userId: number,
  firstName: string,
  lastName: string,
  gender: Gender,
  dateOfBirth: string,
  email: string,
  enrollments: Enrollment[]

}

export interface StudentRequest {

  firstName: string;
  lastName: string;
  gender: Gender;
  dateOfBirth: string;
  email: string
}
